lib/helpers-get-urls/get-urls-from-file-online.js

```diff
@@ -0,0 +1,58 @@
+'use strict'
+
+const path = require('path')
+const https = require('https')
+const clc = require('cli-color')
+const parseFile = require('./parse-file')
+const Cache = require('../Cache')
+
+module.exports = (url, fileType, cacheTime) => {
+  return new Promise(async function (resolve, reject) {
+    const urlHash = require('crypto').createHash('sha1').update(url).digest('hex')
+
+    var greenOnBlack = clc.xterm(47).bgXterm(0)
+    var cachePath = path.resolve(`${__dirname}/../../cache/sitemap/`)
+    var cache = new Cache(urlHash, cachePath, cacheTime)
+    var cacheOld = cache.getKey(url)
+
+    if (cacheTime !== 0 && cacheOld !== undefined) {
+      console.log('\n' + greenOnBlack('Success') + ` Online File Cache found for ${url}`)
+      console.log('\n' + 'Pages found in cache')
+      cacheOld.forEach(e => { console.log(greenOnBlack(e)) })
+      return resolve(cacheOld)
+    } else {
+      console.log(`\nFetching ${url} ...`)
+
+      /*
+      https://www.twilio.com/blog/2017/08/http-requests-in-node-js.html
+       */
+      var data = ''
+      https.get(url, (resp) => {
```

Comment:
I recommend moving this to it's own module. Wrap it in a promise and you use await in here.

Reply:
Will do